packages/flutter_tools/lib/src/devfs.dart

```diff
@@ -518,6 +518,20 @@ class DevFS {
     // dill files that depend on the invalidated files.
     _logger.printTrace('Compiling dart to kernel with ${invalidatedFiles.length} updated files');
 
+    // `generated_main.dart` contains the Dart plugin registry.
+    if (projectRootPath != null) {
```

Comment:
In the original PR wasn't this logic in the resident compiler? Is there an advantage of hoisting this into the devfs?

This condition is also not currently unit tested, there are some test cases in https://github.com/flutter/flutter/blob/master/packages/flutter_tools/test/general.shard/devfs_test.dart you could copy from but that might be a bit more awkward.

Reply:
Moved to the the resident compiler. Done.